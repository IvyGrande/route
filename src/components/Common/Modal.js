import React, {useState} from "react";
import "../../App.css"
import uuid from 'react-uuid';

export const Modal = (props) => {
    const [url, setUrl] = useState(props.url);
    const [name, setName] = useState(props.name);
    const [introduction, setIntroduction] = useState(props.introduction);

    const handleUrlChange = (e) => {
        setUrl(e.target.value)
    };

    const handleNameChange = (e) => {
        setName(e.target.value)
    };

    const handleIntroductionChange = (e) => {
        setIntroduction(e.target.value)
    };

    const cancel = () => {
        props.cancel(false)
    }

    const save = () => {
        const newProduct = {id: uuid(), url, name, introduction}
        props.addNewProduct(newProduct, false, props.id)
    }

    return (
        <div className="addPage">
            <div className="head">{props.modalName}</div>
            <div className="textBlock">
                <p>
                    avatar url :
                    <input type="text" name="url" value={url} onChange={handleUrlChange}/>
                </p>
                <p>
                    product name :
                    <input type="text" name="name" value={name} onChange={handleNameChange}/>
                </p>
                <p>
                    desc :
                    <input type="text" name="introduction" value={introduction}
                           onChange={handleIntroductionChange}/>
                </p>
            </div>
            <div className="bottom">
                <button onClick={cancel}>Cancel</button>
                <button onClick={save}>
                    Ok
                </button>
            </div>
        </div>
    )
}
