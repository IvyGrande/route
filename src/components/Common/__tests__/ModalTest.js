import Enzyme, {mount, shallow} from 'enzyme'
import Adapter from '@cfaester/enzyme-adapter-react-18';
import {Modal} from "../Modal";

Enzyme.configure({adapter: new Adapter()})
describe('Modal component', () => {
    it('should render correct components', () => {
        const wrapper = shallow(<Modal/>);

        expect(wrapper.find('.addPage').length).toBe(1);
        expect(wrapper.find('.bottom').length).toBe(1);
    });

    it('simulates click cancel button', () => {
        const cancelClickBtn = jest.fn();
        const wrapper = shallow(<Modal cancel={cancelClickBtn}/>);
        const cancelBtn = wrapper.find('button').at(0);

        cancelBtn.simulate('click');
        expect(cancelClickBtn).toHaveBeenCalled();
    });


    it('Should set url to state when input is changed', () => {
        const container = shallow(<Modal/>);
        const input = container.find('input').at(0);
        input.simulate('change', {target: {value: "url"}});
        expect(container).toMatchSnapshot();
    });
    exports[`Index with enzyme Should set value to state when input is changed 1`] = `
  <input
    onChange={[Function]}
    type="text"
    value="url"
  />
`;

    it('Should set name to state when input is changed', () => {
        const container = shallow(<Modal/>);
        const input = container.find('input').at(0);
        input.simulate('change', {target: {value: "name"}});
        expect(container).toMatchSnapshot();
    });
    exports[`Index with enzyme Should set value to state when input is changed 1`] = `
  <input
    onChange={[Function]}
    type="text"
    value="name"
  />
`;

    it('Should set introduction to state when input is changed', () => {
        const container = shallow(<Modal/>);
        const input = container.find('input').at(0);
        input.simulate('change', {target: {value: "introduction"}});
        expect(container).toMatchSnapshot();
    });
    exports[`Index with enzyme Should set value to state when input is changed 1`] = `
  <input
    onChange={[Function]}
    type="text"
    value="introduction"
  />
`;
})