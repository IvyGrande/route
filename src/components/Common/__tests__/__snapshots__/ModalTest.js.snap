// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Should set introduction to state when input is changed 1`] = `ShallowWrapper {}`;

exports[`Modal component Should set name to state when input is changed 1`] = `ShallowWrapper {}`;

exports[`Modal component Should set url to state when input is changed 1`] = `ShallowWrapper {}`;

exports[`deleteAlert component Should set introduction to state when input is changed 1`] = `ShallowWrapper {}`;

exports[`deleteAlert component Should set name to state when input is changed 1`] = `ShallowWrapper {}`;

exports[`deleteAlert component Should set url to state when input is changed 1`] = `ShallowWrapper {}`;

exports[`deleteAlert component Should set value to state when input is changed 1`] = `ShallowWrapper {}`;
